src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -175,39 +176,142 @@ private BoundPattern BindPropertyPattern(PropertyPatternSyntax node, BoundExpres
         private ImmutableArray<BoundSubPropertyPattern> BindSubPropertyPatterns(PropertyPatternSyntax node, TypeSymbol type, DiagnosticBag diagnostics)
         {
             var result = ArrayBuilder<BoundSubPropertyPattern>.GetInstance();
-            foreach (var syntax in node.PatternList.SubPatterns)
+            foreach (var e in node.Expressions)
             {
-                var propName = syntax.Left;
-                BoundPattern pattern;
-                HashSet<DiagnosticInfo> useSiteDiagnostics = null;
-                LookupResultKind resultKind;
-                Symbol property = FindPropertyOrFieldByName(type, propName, out resultKind, ref useSiteDiagnostics);
-                if ((object)property != null)
+                var syntax = e as IsPatternExpressionSyntax;
```

Comment:
You told me you prefer an "unexpected" exception rather than a CastException, so I left the `as` in the code and added an exception, below.